dev/src/bulk-writer.ts

```diff
@@ -817,7 +879,22 @@ export class BulkWriter {
       this._errorFn.bind(this),
       this._successFn.bind(this)
     );
-    this._sendFn(enqueueOnBatchCallback, bulkWriterOp);
+    if (this._pendingOpsCount < this._maxPendingOpCount) {
+      this._pendingOpsCount++;
+      this._sendFn(enqueueOnBatchCallback, bulkWriterOp);
+    } else {
+      // Increment the _lastOp pointer to ensure that the promise returned by
+      // flush() includes this operation.
+      this._lastOp = this._lastOp.then(() => {
```

Comment:
I think we should do this in both cases and remove the duplication between here and _sendFn.